### ISSUE TITLE
Is there a preferred way how 3rd party projects can check option used for tesseract build

### ISSUE THREAD

**Issue by zdenop** [tesseract-ocr/tesseract#2372]
Is there a preferred way how 3rd party projects can check whether `disable-legacy` was used for tesseract build:
1. during build/configure of project?
2. at runtime?

This question is not related only to this build options, but IMO should be answered during committing this kind of changes.

_Originally posted by @zdenop in https://github.com/tesseract-ocr/tesseract/pull/2368#issuecomment-479888070_

**Comment by amitdo** [comment]
The disable-legacy build option was added by me so a future tesseract version can easily drop the legacy code.

See #707 for the context.

Currently, there is no way for 3rd party programs or libraries to discover that this build option was used.

I think both run-time and compile time discovery options are needed.

**Comment by amitdo** [comment]
>during build/configure of project?

I think the proper way to do it is to add -DDISABLED_LEGACY_ENGINE to the Cflags in the .pc file.

**Comment by amitdo** [comment]
>at runtime?

for this one we can add:

bool get_build_option(char* option);

**Comment by zdenop** [comment]
And what about to put such information (build option having effect on available features e.g. DISABLE_GRAPHICS) to tess_version.h (it is generated file from from [tess_version.h.in](https://github.com/tesseract-ocr/tesseract/blob/master/src/api/tess_version.h.in)?

**Comment by amitdo** [comment]
Also possible.

Did you see this:
https://github.com/tesseract-ocr/tesseract/issues/2372#issuecomment-480559819
?

**Comment by zdenop** [comment]
Yes, I saw it. pck_config is only one of ways, how to get configuration (and IMO limited to linux infrastructure, but it need to provided if we support pck_config). 
So we need to have more universal approach.
